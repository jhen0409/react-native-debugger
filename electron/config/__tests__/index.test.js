import fs from 'fs';
import path from 'path';

jest.mock('electron', () => ({
  shell: {
    openItem: jest.fn(),
  },
}));

const testFile = path.join(__dirname, 'config_test');

beforeAll(() => fs.existsSync(testFile) && fs.unlinkSync(testFile));

/* eslint-disable global-require */
test('readConfig', () => {
  const { readConfig } = require('..');

  expect(readConfig(testFile)).toMatchSnapshot();

  // User custom config
  fs.writeFileSync(testFile, '{ autoUpdate: false, }');
  expect(readConfig(testFile)).toMatchSnapshot();

  // Broken config
  fs.writeFileSync(testFile, '{ autoUpdate: is_broken, }');
  expect(readConfig(testFile)).toMatchSnapshot();
});

test('openConfigFile', () => {
  const { readConfig, openConfigFile } = require('..');
  const { shell } = require('electron');

  openConfigFile(testFile);
  expect(shell.openItem).toBeCalledWith(testFile);
  shell.openItem.mockClear();

  fs.unlinkSync(testFile);
  openConfigFile(testFile);
  expect(readConfig(testFile)).toMatchSnapshot();
});
