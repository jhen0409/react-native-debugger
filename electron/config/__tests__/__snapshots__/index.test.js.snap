// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`openConfigFile 1`] = `
Object {
  "config": Object {
    "autoUpdate": true,
    "defaultNetworkInspect": false,
    "defaultRNPackagerPorts": Array [
      8081,
    ],
    "defaultReactDevToolsPort": 19567,
    "defaultReactDevToolsTheme": "RNDebugger",
    "editor": "",
    "timesJSLoadToRefreshDevTools": -1,
    "windowBounds": Object {
      "frame": true,
    },
  },
}
`;

exports[`readConfig 1`] = `
Object {
  "config": Object {
    "autoUpdate": true,
    "defaultNetworkInspect": false,
    "defaultRNPackagerPorts": Array [
      8081,
    ],
    "defaultReactDevToolsPort": 19567,
    "defaultReactDevToolsTheme": "RNDebugger",
    "editor": "",
    "timesJSLoadToRefreshDevTools": -1,
    "windowBounds": Object {
      "frame": true,
    },
  },
}
`;

exports[`readConfig 2`] = `
Object {
  "config": Object {
    "autoUpdate": false,
  },
}
`;

exports[`readConfig 3`] = `
Object {
  "config": Object {
    "autoUpdate": true,
    "defaultNetworkInspect": false,
    "defaultRNPackagerPorts": Array [
      8081,
    ],
    "defaultReactDevToolsPort": 19567,
    "defaultReactDevToolsTheme": "RNDebugger",
    "editor": "",
    "timesJSLoadToRefreshDevTools": -1,
    "windowBounds": Object {
      "frame": true,
    },
  },
  "error": [SyntaxError: Unexpected 'i' at line 1 column 16 of the JSON5 data. Still to read: "is_broken, }"],
  "isConfigBroken": true,
}
`;
